proposed/2021/NuGet.orgTFMs.md

```diff
@@ -0,0 +1,96 @@
+
+### Success: How do we know if we’ve solved this problem? 
+
+- By measuring the change in the percentage of successful installs and lowering the amount of [NU1202](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1202) errors in telemetry. 
+- By measuring the usage of target framework & supported platform filters on NuGet.org & Visual Studio. 
+- By gauging excitement/disappointment after blogging about this feature & analyzing sentiment on Twitter/GitHub/DevCom/etc. 
+
+### Audience: Who are we building for? 
+
+We are building this for the .NET developer who is browsing & making decisions about including a NuGet package into their project based on target framework & supported platform criteria. 
+
+## Explanation
+
+### Functional Explanation
+
+#### Context and Scope 
+When a developer accesses the details page for a NuGet package, they will see information on target frameworks in a dropdown menu. The framework information will be surfaced from the package. In cases where versions are not specified (such as .net) or other unexpected scenarios, ‘Version Not Specified’ or similar messages will be surfaced. 
+
+##### Minimal Requirements 
+- List of Frameworks on NuGet.org
+
+##### Non-Requirements 
+- List of Platforms on NuGet.org
+  - Users interviewed did not express need for supported platform information 
+  - Cannot be surfaced directly from package data and would require additional layer of soft calculation from engineering (no clear mapping scheme currently exists between target frameworks and supported platforms) 
+  - Can be revisited in future 
+- Filters for Frameworks on NuGet.org
+    - Users interviewed were ambivalent on filters, requires further investigation. Can be revisited in future 
+- Filters for Platforms on NuGet.org
+  - Users interviewed were ambivalent on filters, requires further investigation. Can be revisited in future 
+
+Although Target Frameworks & Supported Platforms are relatively similar in general contexts and often have overlap, we must clearly define each of these items based on their differences. Today we live in an ecosystem where Target Frameworks can represent both a supported platform, or a subset of supported platforms. 
+
+**Target Framework** – A target framework is a specification of the APIs available to an application or library. 
+
+Example: .NET Framework, .NET Core, .NET Standard, .NET 5 are target frameworks. 
+
+**Supported Platform** – A supported platform is a target that the APIs can be ran on. 
+
+Example: iOS, Android, Windows are platforms. 
+
+### Solution 
+
+- A .NET developer can view Frameworks of a package on NuGet.org: 
+![](../../meta/resources/NuGet.orgTFMs/PackageDetailsWithTFMs.png)
+
+### Goals and Non-Goals 
+
+#### Goals 
+
+Make it clear to developers whether a package can be installed or updated for their project or solution based on a target framework. 
```

Comment:
We should be clear whether we are listing all frameworks that are supported _or_ exact frameworks that have assets.

Consider a simple package with `lib/netstandard1.0/foo.dll`. Are we going to just say `netstandard1.0` or are we going to list all of the frameworks that support `netstandard1.0`.

Reply:
just netstandard1.0. we don't care about the whole list b/c that can be looked up by users. We should link to tables for any portable assets though like https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support

Reply:
The logistics concern me there--we would need to maintain a dictionary, which could mean that we have the potential for it to contain errors or fall out of maintenance. Those who use abstracted APIs like netstandard would usually understand the abstraction, but we could provide a link to a conversion matrix a la https://docs.microsoft.com/en-us/dotnet/standard/net-standard, which given its location in docs should be kept up to date.

Reply:
There's the other consideration that portable assets != platform assets & therefore we can't just list supported platforms as we can mislead users.

Reply:
That's an important distinction. I've added in the Technical Explanation section:
```
- When a portable TFM is being broken up and displayed (if feasible), we will need to flag it as portable, as portable assets are not the same as platform assets.
```